### ISSUE TITLE
Model prediction is slow, anyway to speed it up?

### ISSUE THREAD

**Issue by LifeArchitect** [justinshenk/fer#20]

**Comment by justinshenk** [comment]
Yes, we could batch images through the model. Are you using cpu only or also GPU acceleration?

**Comment by Neel7317** [comment]
how can we change FPS for Model prediction?

**Comment by justinshenk** [comment]
Consider setting the [`save_fps`](https://github.com/justinshenk/fer/blob/4ef7087050ddde00cf0fd8024ce08a8279ddd169/src/fer/classes.py#L174) parameter to a number 2-25. This will only compute on the nth frame.

**Comment by Neel7317** [comment]
thanks for answering but after changing save_fps parameter, i have to retrain the model ? and if i give save_fps=3 or 4 then will model get decrease its  accuracy?

**Comment by justinshenk** [comment]
`save_fps` just changes how many frames are skipped when running inference
on the video. It doesn't change the model, only how it is applied to the
video.

**Comment by Neel7317** [comment]
means if i am use this below code then where do i make changes for save_fps can you tell me please ?

def foremotion():
    # fn contains the video file name (Ensure that you only upload one file)
    videofile = "D://RR/integration/in.mp4"
    # Face detection
    detector = FER(mtcnn=True)
    # Video predictions
    video = Video(videofile)

    # Output list of dictionaries
    raw_data = video.analyze(detector, display=False)

    # Convert to pandas for analysis
    df = video.to_pandas(raw_data)
    df = video.get_first_face(df)
    df = video.get_emotions(df)

    # Plot emotions
    fig = df.plot(figsize=(20, 10), fontsize=14).get_figure()
    # Filename for plot
    fig.savefig('static/my_figure.png')

    return render_template('try.html',img="static/my_figure.png")

**Comment by Neel7317** [comment]
means if i am using this below code then where do i make changes for
save_fps can you tell me please ?

def foremotion():
# fn contains the video file name (Ensure that you only upload one file)
videofile = "D://RR/integration/in.mp4"
# Face detection
detector = FER(mtcnn=True)
# Video predictions
video = Video(videofile)

# Output list of dictionaries
raw_data = video.analyze(detector, display=False)

# Convert to pandas for analysis
df = video.to_pandas(raw_data)
df = video.get_first_face(df)
df = video.get_emotions(df)

# Plot emotions
fig = df.plot(figsize=(20, 10), fontsize=14).get_figure()
# Filename for plot
fig.savefig('static/my_figure.png')

return render_template('try.html',img="static/my_figure.png")

**Comment by Neel7317** [comment]
Thank you so much. I just found where to put that and it's working fine.
and another doubt in my mind about whether we can have any solution to make
this entire process fast? I have a GPU too but still it takes too much time
for complete execution for just 1 minute video file.

**Comment by justinshenk** [comment]
It could be extended for this, maybe someone else will be kind enough to
contribute this feature.
